lib/logger-factory.js

```diff
@@ -1,88 +1,89 @@
 'use strict'
 
-/**
- * Code imported from `pino-http`
- * Repo: https://github.com/pinojs/pino-http
- * License: MIT (https://raw.githubusercontent.com/pinojs/pino-http/master/LICENSE)
- */
-
-const nullLogger = require('abstract-logging')
-const pino = require('pino')
-const { serializersSym } = pino.symbols
 const {
-  FST_ERR_LOG_INVALID_DESTINATION,
-  FST_ERR_LOG_INVALID_LOGGER,
-  FST_ERR_LOG_INVALID_LOGGER_INSTANCE,
+  FST_ERR_LOG_LOGGER_AND_LOGGER_INSTANCE_PROVIDED,
   FST_ERR_LOG_INVALID_LOGGER_CONFIG,
-  FST_ERR_LOG_LOGGER_AND_LOGGER_INSTANCE_PROVIDED
+  FST_ERR_LOG_INVALID_LOGGER_INSTANCE,
+  FST_ERR_LOG_INVALID_LOGGER
 } = require('./errors')
 
-function createPinoLogger (opts) {
-  if (opts.stream && opts.file) {
-    throw new FST_ERR_LOG_INVALID_DESTINATION()
-  } else if (opts.file) {
-    // we do not have stream
-    opts.stream = pino.destination(opts.file)
-    delete opts.file
+/**
+ * Utility for creating a child logger with the appropriate bindings, logger factory
+ * and validation.
+ * @param {object} context
+ * @param {import('../fastify').FastifyBaseLogger} logger
+ * @param {import('../fastify').RawRequestDefaultExpression<any>} req
+ * @param {string} reqId
+ * @param {import('../types/logger.js').ChildLoggerOptions?} loggerOpts
+ *
+ * @returns child logger instance
```

Comment:
This is not the correct format for `@returns`. At a minimum, it should be:
```suggestion
 * @returns {object} New logger instance, inheriting all parent bindings,
 * with child bindings added.
```